workers/unity/Packages/com.improbable.gdk.core/Config/ConnectionConfig.cs

```diff
@@ -5,14 +5,52 @@ namespace Improbable.Gdk.Core
 {
     public abstract class ConnectionConfig
     {
+        /// <summary>
+        ///     The type of networking that should be used.
+        /// </summary>
+        /// <remarks>
+        ///    Default is RakNet.
+        /// </remarks>
         public NetworkConnectionType LinkProtocol = NetworkConnectionType.RakNet;
+
+        /// <summary>
+        ///     Denotes whether protocol logging should be enabled.
+        /// </summary>
+        /// <remarks>
+        ///    Default is false.
+        /// </remarks>
         public bool EnableProtocolLoggingAtStartup = false;
+
+        /// <summary>
+        ///     Denotes whether the worker uses the local IP to connect to the SpatiaLOS runtime.
+        /// </summary>
+        /// <remarks>
+        ///    Default is false.
+        ///    This should be true if connecting client-workers to cloud deployments or if connecting via the
+        ///    locator flow.
+        /// </remarks>
         public bool UseExternalIp = false;
+
+        /// <summary>
+        ///     The ID of the worker.
+        /// </summary>
         public string WorkerId;
+
+        /// <summary>
+        ///     The type of the worker.
+        /// </summary>
         public string WorkerType;
 
+        /// <summary>
+        ///    Validates that the connection configuration is sane.
```

Comment:
What does "sane" mean? And what happens if it's not? This function signature is not what I'd expect for this so it will need a longer comment unless we consider this purely internal. (It also means we need to change this probably).

Reply:
I'm tempted to just remove this doc string

Reply:
Abstract functions like this shouldn't rely on the user implementation throwing as part of their control flow. Although that is an issue for later.